var/spack/repos/builtin/packages/orca/package.py

```diff
@@ -50,15 +55,27 @@ class Orca(Package):
     depends_on("libpciaccess", type="run")
 
     # Map Orca version with the required OpenMPI version
-    openmpi_versions = {"4.0.1.2": "2.0.2", "4.2.0": "3.1.4", "4.2.1": "3.1.4", "5.0.3": "4.1.2"}
+    openmpi_versions = {
+        "4.0.1.2": "2.0.2",
+        "4.2.0": "3.1.4",
+        "4.2.1": "3.1.4",
+        "5.0.3": "4.1.1",
+        "5.0.4": "4.1.1",
+    }
     for orca_version, openmpi_version in openmpi_versions.items():
         depends_on(
             "openmpi@{0}".format(openmpi_version), type="run", when="@{0}".format(orca_version)
         )
 
     def url_for_version(self, version):
-        out = "file://{0}/orca_{1}_linux_x86-64_openmpi{2}.tar.zst"
-        return out.format(os.getcwd(), version.underscored, self.openmpi_versions[version.string])
+        openmpi_version = "".join(self.openmpi_versions[version.string].split("."))
+        url = "file://{0}/orca_{1}_linux_x86-64_shared_openmpi{2}.tar.xz"
+
+        if self.spec.satisfies("@:4.2.1"):
+            openmpi_version = self.openmpi_versions[version.string]
+            url = "file://{0}/orca_{1}_linux_x86-64_openmpi{2}.tar.zst"
```

Comment:
FYI. We encourage switching from format to Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) in packages, when feasible, as we transition to them within the core.

Reply:
If you make this change you could tighten the contents of the method.